pkg/ebpf/probes/probes.go

```diff
@@ -2,174 +2,486 @@ package probes
-			SecurityMmapFile:           {event: "security_mmap_file", probeType: kprobe, fn: "trace_security_mmap_file"},
-			DoSplice:                   {event: "do_splice", probeType: kprobe, fn: "trace_do_splice"},
-			DoSpliceRet:                {event: "do_splice", probeType: kretprobe, fn: "trace_ret_do_splice"},
-			ProcCreate:                 {event: "proc_create", probeType: kprobe, fn: "trace_proc_create"},
-			SecurityFileMProtect:       {event: "security_file_mprotect", probeType: kprobe, fn: "trace_security_file_mprotect"},
-			CommitCreds:                {event: "commit_creds", probeType: kprobe, fn: "trace_commit_creds"},
-			SwitchTaskNS:               {event: "switch_task_namespaces", probeType: kprobe, fn: "trace_switch_task_namespaces"},
-			ARMKprobe:                  {event: "arm_kprobe", probeType: kprobe, fn: "trace_arm_kprobe"},
-			CallUsermodeHelper:         {event: "call_usermodehelper", probeType: kprobe, fn: "trace_call_usermodehelper"},
-			DebugfsCreateFile:          {event: "debugfs_create_file", probeType: kprobe, fn: "trace_debugfs_create_file"},
-			DebugfsCreateDir:           {event: "debugfs_create_dir", probeType: kprobe, fn: "trace_debugfs_create_dir"},
-			DeviceAdd:                  {event: "device_add", probeType: kprobe, fn: "trace_device_add"},
-			RegisterChrdev:             {event: "__register_chrdev", probeType: kprobe, fn: "trace___register_chrdev"},
-			RegisterChrdevRet:          {event: "__register_chrdev", probeType: kretprobe, fn: "trace_ret__register_chrdev"},
-			DoInitModule:               {event: "do_init_module", probeType: kprobe, fn: "trace_do_init_module"},
-			DoInitModuleRet:            {event: "do_init_module", probeType: kretprobe, fn: "trace_ret_do_init_module"},
-			LoadElfPhdrs:               {event: "load_elf_phdrs", probeType: kprobe, fn: "trace_load_elf_phdrs"},
-			Filldir64:                  {event: "filldir64", probeType: kprobe, fn: "trace_filldir64"},
-			TaskRename:                 {event: "task:task_rename", probeType: rawTracepoint, fn: "tracepoint__task__task_rename"},
-		},
+		probes: allProbes,
 		module: module,
 	}, nil
 }
 
-//Attach the probe corresponding to the probe's handle
-func (p *probes) Attach(probe Handle) error {
-	if p.probes[probe] == nil {
-		return fmt.Errorf("failed to attach probe: doesn't exist")
+func (p *probes) Attach(handle Handle, args ...interface{}) error {
+	return p.attachOrDetach(handle, true, args...)
+}
+
+func (p *probes) Detach(handle Handle, args ...interface{}) error {
+	return p.attachOrDetach(handle, false, args...)
+}
+
+func (p *probes) attachOrDetach(handle Handle, attach bool, args ...interface{}) error {
+	var netIface *net.Interface
+
+	for _, arg := range args {
+		switch a := arg.(type) {
+		case *net.Interface:
+			netIface = a
+		}
+	}
+
+	if _, ok := p.probes[handle]; !ok {
+		return fmt.Errorf("incorrect event handle provided")
 	}
 
-	return p.probes[probe].attach(p.module)
+	switch b := p.probes[handle].(type) {
```

Comment:
So since we use (args ...interface{}) Do we actually need this switch? Internally the functions should interpret if and what args are relevant no? I can see leaving it as a safety measure though so up to you.